soc/xtensa/intel_adsp/common/ipc.c

```diff
@@ -206,6 +207,19 @@ void intel_adsp_ipc_send_message_emergency(const struct device *dev, uint32_t da
 	regs->idr = data | INTEL_ADSP_IPC_BUSY;
 }
 
+void intel_adsp_ipc_force_dma_l0_state(void)
+{
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE
+	ACE_DfPMCCH.svcfg |= ADSP_FORCE_DECOUPLED_HDMA_L1_EXIT_BIT;
+#endif
+}
+
+void intel_adsp_ipc_allow_dma_l1_state(void)
+{
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE
+	ACE_DfPMCCH.svcfg &= ~(ADSP_FORCE_DECOUPLED_HDMA_L1_EXIT_BIT);
+#endif
+}
```

Comment:
The code looks otherwise good, but I'm not sure about the correct location. This is really a SoC-wide quirk that affects the HD DMA operation. So having it in soc/xtensa/intel_adsp/ makes sense, so this part is correct. But this doesn't really have much to do with ipc, so I'd rather put this into soc/xtensa/intel_adsp/common/include/intel_adsp_hda.h 
... and keep it a no-op for non-ace platforms.

Reply:
I moved the functions to the file you mentioned.